heron/schedulers/src/java/com/twitter/heron/scheduler/aurora/AuroraScheduler.java

```diff
@@ -184,9 +185,29 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
     return true;
   }
 
+
+  boolean confirmWithUser(int newContainerCount) {
+    LOG.info(String.format("After update there will be %d more containers. "
+        + "Please make sure there are sufficient resources to update this job. "
+        + "Continue update? [y/N]: ", newContainerCount));
+    Scanner scanner = new Scanner(System.in);
+    String userInput = scanner.nextLine();
+    if ("y".equalsIgnoreCase(userInput)) {
+      return true;
+    }
+    return false;
+  }
+
   @Override
   public Set<PackingPlan.ContainerPlan> addContainers(
       Set<PackingPlan.ContainerPlan> containersToAdd) {
+    if ("prompt".equalsIgnoreCase(Context.updatePrompt(config))) {
+      if (!confirmWithUser(containersToAdd.size())) {
```

Comment:
try to avoid if else or if if, maybe single statement is a little bit clear. But I am not sure the entire logic, I could be wrong.

```
Boolean hasNotConfirmedWithUser = "prompt".equalsIgnoreCase(Context.updatePrompt(config)) && !hasConfirmedWithUser(containersToAdd.size());
  if(hasNotConfirmedWithUser) LOG.warning("Scheduler updated topology canceled.");
```

Reply:
updated